File: app/src/main/java/tech/salroid/filmy/Database/MovieDetailsUpdation.java
package tech.salroid.filmy.database;

import android.content.ContentValues;
import android.content.Context;
import java.util.HashMap;


public class MovieDetailsUpdation {

    public static void performMovieDetailsUpdation(Context context, int type, HashMap<String,String> movieMap,String movie_id){


        ContentValues contentValues = new ContentValues();

        contentValues.put(FilmContract.MoviesEntry.MOVIE_BANNER, movieMap.get("banner"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_TAGLINE, movieMap.get("tagline"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_DESCRIPTION, movieMap.get("overview"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_TRAILER, movieMap.get("trailer_img"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_CERTIFICATION, movieMap.get("certification"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_LANGUAGE, movieMap.get("language"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RUNTIME, movieMap.get("runtime"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RELEASED, movieMap.get("released"));
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RATING, movieMap.get("rating"));

        switch (type) {

            case 0:

                final String selection =
                        FilmContract.MoviesEntry.TABLE_NAME +
                                "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? ";
                final String[] selectionArgs = {movie_id};

                long id = context.getContentResolver().update(FilmContract.MoviesEntry.buildMovieByTag(movie_id), contentValues, selection, selectionArgs);

                if (id != -1) {
                    //  Log.d(LOG_TAG, "Movie row updated with new values.");
                }

                break;

            case 1:

                final String selection2 =
                        FilmContract.InTheatersMoviesEntry.TABLE_NAME +
                                "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? ";
                final String[] selectionArgs2 = {movie_id};

                long id2 = context.getContentResolver().update(FilmContract.InTheatersMoviesEntry.buildMovieByTag(movie_id), contentValues, selection2, selectionArgs2);

                if (id2 != -1) {
                    //  Log.d(LOG_TAG, "Movie row updated with new values.");
                }
                break;

            case 2:


                final String selection3 =
                        FilmContract.UpComingMoviesEntry.TABLE_NAME +
                                "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? ";
                final String[] selectionArgs3 = {movie_id};

                long id3 = context.getContentResolver().update(FilmContract.UpComingMoviesEntry.buildMovieByTag(movie_id), contentValues, selection3, selectionArgs3);

                if (id3 != -1) {
                    //  Log.d(LOG_TAG, "Movie row updated with new values.");
                }

                break;

        }


    }
}

File: app/src/main/java/tech/salroid/filmy/database/MovieDetailsUpdation.kt
package tech.salroid.filmy.database

import android.content.ContentValues
import android.content.Context

object MovieDetailsUpdate {

    fun performMovieDetailsUpdate(
        context: Context,
        type: Int,
        movieMap: MutableMap<String, String?>,
        movieId: String?
    ) {

        val contentValues = ContentValues()
        contentValues.put(FilmContract.MoviesEntry.MOVIE_BANNER, movieMap["banner"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_TAGLINE, movieMap["tagline"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_DESCRIPTION, movieMap["overview"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_TRAILER, movieMap["trailer_img"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_CERTIFICATION, movieMap["certification"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_LANGUAGE, movieMap["language"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RUNTIME, movieMap["runtime"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RELEASED, movieMap["released"])
        contentValues.put(FilmContract.MoviesEntry.MOVIE_RATING, movieMap["rating"])

        when (type) {
            0 -> {
                val selection = FilmContract.MoviesEntry.TABLE_NAME +
                        "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? "
                val selectionArgs = arrayOf(movieId)

                val id = context.contentResolver.update(
                    FilmContract.MoviesEntry.buildMovieByTag(movieId),
                    contentValues,
                    selection,
                    selectionArgs
                ).toLong()

                /* if (id != -1L) {
                     Log.d(LOG_TAG, "Movie row updated with new values.");
                 }*/
            }
            1 -> {
                val selection2 = FilmContract.InTheatersMoviesEntry.TABLE_NAME +
                        "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? "
                val selectionArgs2 = arrayOf(movieId)

                val id2 = context.contentResolver.update(
                    FilmContract.InTheatersMoviesEntry.buildMovieByTag(movieId),
                    contentValues,
                    selection2,
                    selectionArgs2
                ).toLong()

                /* if (id2 != -1L) {
                     Log.d(LOG_TAG, "Movie row updated with new values.");
                 }*/
            }
            2 -> {

                val selection3 = FilmContract.UpComingMoviesEntry.TABLE_NAME +
                        "." + FilmContract.MoviesEntry.MOVIE_ID + " = ? "
                val selectionArgs3 = arrayOf(movieId)
                val id3 = context.contentResolver.update(
                    FilmContract.UpComingMoviesEntry.buildMovieByTag(movieId),
                    contentValues,
                    selection3,
                    selectionArgs3
                ).toLong()

                /*if (id3 != -1L) {
                     Log.d(LOG_TAG, "Movie row updated with new values.");
                }*/
            }
        }
    }
}